core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -766,6 +785,29 @@ class RequestMetrics(name: String) extends KafkaMetricsGroup {
     }
   }
 
+  class Histogram(val metricNamePrefix: String, val metricNamePostfix: String, val binBoundaries: Array[Int]) {
+    // binLeftBoundary => (name, Counter)
+    val boundaryCounterMap = createHistogram()
+
+    def update(value: Double): Unit = {
```

Comment:
I might be wrong, but it seems like if someone is logging an int, that means it goes Double -> Long -> Int. Given the goal above of using an int instead of a long, this seems like it will negate any compute benefit of using an int instead of a long.

Is there a way to provide an overload here? Or maybe updateInt updateDouble etc?

Reply:
added more update methods to support different value types